File: tests/fixtures/vulnerable/py/incomplete-security.py
# Incomplete Security Implementation (Python)
# Should trigger: security-todo, commented-security-check, placeholder-credentials, incomplete-error-handling

from flask import Flask, request, jsonify
import jwt

app = Flask(__name__)

# Issue 1: Security TODOs
# TODO: Add authentication middleware
# TODO: Implement password validation
def login(username, password):
    # FIXME: Need to hash passwords before comparing
    return authenticate_user(username, password)

# Issue 2: Commented out security validation
def delete_resource(resource_id, user):
    # if not user.has_permission('delete'):
    #     return {'error': 'Forbidden'}, 403
    
    # Dangerous operation without permission check!
    return db.delete(resource_id)

# Issue 3: Placeholder credentials
API_KEY = "YOUR_API_KEY_HERE"
DB_PASSWORD = "changeme"
SECRET_TOKEN = "placeholder-token"
ADMIN_USER = "admin"
ADMIN_PASS = "test123"

JWT_SECRET = "PUT_YOUR_SECRET_HERE"
ENCRYPTION_KEY = "ENTER_YOUR_KEY"

# Issue 4: Incomplete error handling
def verify_token(token):
    try:
        decoded = jwt.decode(token, JWT_SECRET, algorithms=['HS256'])
        return decoded
    except Exception:
        pass  # Empty except - silently fails!

# Another incomplete error handler
def authenticate(credentials):
    try:
        return check_credentials(credentials)
    except Exception as e:
        # TODO: Implement proper error handling
        print(e)

# More security TODOs
def store_sensitive_data(data):
    # TODO: Add encryption before storing
    # FIXME: Validate input data for security
    return db.insert(data)

# Commented out authorization check
@app.route('/admin/delete', methods=['POST'])
def admin_delete():
    # if not request.user.is_admin:
    #     return jsonify({'error': 'Unauthorized'}), 401
    
    delete_all_data()  # Dangerous!
    return jsonify({'success': True})

if __name__ == '__main__':
    app.run(debug=True)

File: tests/fixtures/vulnerable/js/incomplete-security.js
// Incomplete Security Implementation (JavaScript)
// Should trigger: security-todo, commented-security-check, placeholder-credentials, incomplete-error-handling

const express = require('express');
const app = express();

// Issue 1: Security TODO
// TODO: Add proper authentication before production
// TODO: Implement password strength validation
function login(username, password) {
  // FIXME: Need to add rate limiting for security
  return authenticateUser(username, password);
}

// Issue 2: Commented out security check
function deleteUser(userId, currentUser) {
  // if (!currentUser.hasPermission('admin')) {
  //   return { error: 'Forbidden' };
  // }
  
  // Dangerous operation without permission check!
  return db.delete('users', userId);
}

// Issue 3: Placeholder credentials
const config = {
  apiKey: "YOUR_API_KEY_HERE",
  dbPassword: "changeme",
  secretToken: "placeholder-token",
  adminUser: "admin",
  adminPass: "test123"
};

const API_KEY = "PUT_YOUR_KEY_HERE";
const SECRET = "ENTER_YOUR_SECRET";

// Issue 4: Incomplete error handling
async function verifyToken(token) {
  try {
    const decoded = await jwt.verify(token, SECRET_KEY);
    return decoded;
  } catch (e) {
    // TODO: Add proper error handling
    console.log(e);
  }
}

// Another incomplete error handler
async function authenticate(credentials) {
  try {
    return await checkCredentials(credentials);
  } catch (err) {
    // Empty catch - silently fails!
  }
}

// More security TODOs
// FIXME: Add encryption before storing sensitive data
function storeUserData(data) {
  // TODO: Hash password before storing
  return db.insert(data);
}

module.exports = { login, deleteUser, verifyToken, authenticate, storeUserData };
